src/plugins/unified_search/public/utils/helpers.ts

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import type { IFieldType } from '@kbn/data-views-plugin/common';
+import { i18n } from '@kbn/i18n';
+import { isEmpty } from 'lodash';
+import { validateParams } from '../filter_bar/filter_editor/lib/filter_editor_utils';
+
+export const getFieldValidityAndErrorMessage = (
+  field: IFieldType,
+  value?: string | undefined
+): { isInvalid: boolean; errorMessage: string[] } => {
+  const type = field.type;
+  switch (type) {
+    case 'string':
+      return {
+        isInvalid: !validateParams(value, field),
+        errorMessage: [''],
+      };
+    case 'date':
+    case 'date_range':
+      return {
+        isInvalid: !isEmpty(value) && !validateParams(value, field),
+        errorMessage: [
+          i18n.translate('unifiedSearch.filter.filterBar.invalidDateFormatProvidedErrorMessage', {
+            defaultMessage: 'Invalid date format provided',
+          }),
+        ],
+      };
+    case 'ip':
+    case 'ip_range':
+      return {
+        isInvalid: !isEmpty(value) && !validateParams(value, field),
+        errorMessage: [''],
```

Comment:
Please explain how errorMessage: [''] works in case of error. I see you did it for multiple cases and it's not clear to me. Does it mean that we set the invalid status without any message?

Reply:
Current issue add an input error display for the Date type. I also added the ability to show an error for other possible cases. In case we want to show error messages for other types. To do this, we will need to add only the text of the error message.

Reply:
Wrong, please don't add code for the future. This is more confusing. About that case, I see you set the validity but no error message. Reading this code, I suspect that we had a requirement to hide the message, which is not the true....

Reply:
Oh. Sure. You are right. I overdid it a little. I will set the check that corresponds to the current issue.